launcher/tests/test_updater.py

```diff
@@ -61,82 +61,81 @@ def test_updater_templatevms_present():
 ):
-    upgrade_generator = Updater.apply_updates(["dom0"])
-    results = {}
-
-    for vm, progress, result in upgrade_generator:
-        results[vm] = result
-        assert progress is not None
-
-    assert Updater.overall_update_status(results) == UpdateStatus.UPDATES_OK
+    assert Updater.apply_updates_dom0() == UpdateStatus.UPDATES_OK
     assert not mocked_error.called
     # We check for updates, but do not attempt to apply them
     check_dom0.assert_called_once_with()
     assert not apply_dom0.called
-    assert not apply_vm.called
 
 
 @mock.patch("sdw_updater.Updater._write_updates_status_flag_to_disk")
 @mock.patch("sdw_updater.Updater._write_last_updated_flags_to_disk")
-@mock.patch(
-    "sdw_updater.Updater._apply_updates_vm",
-    side_effect=[UpdateStatus.UPDATES_OK, UpdateStatus.UPDATES_REQUIRED],
-)
-@mock.patch("sdw_updater.Updater._apply_updates_dom0")
+@mock.patch("sdw_updater.Updater._start_qubes_updater_proc")
 @mock.patch("sdw_updater.Updater.sdlog.error")
 @mock.patch("sdw_updater.Updater.sdlog.info")
-def test_apply_updates_required(
-    mocked_info, mocked_error, apply_dom0, apply_vm, write_updated, write_status
-):
-    upgrade_generator = Updater.apply_updates(["fedora", "sd-app"])
-    results = {}
-
-    for vm, progress, result in upgrade_generator:
-        results[vm] = result
-        assert progress is not None
+def test_apply_templates_success(mocked_info, mocked_error, mock_proc, write_updated, write_status):
+    result = Updater.apply_updates_templates(templates=[])
+    mock_proc.assert_called_once()
+    assert result == UpdateStatus.UPDATES_OK
+    assert not mocked_error.called
 
-    assert results == {"fedora": UpdateStatus.UPDATES_OK, "sd-app": UpdateStatus.UPDATES_REQUIRED}
-    calls = [call("fedora"), call("sd-app")]
-    apply_vm.assert_has_calls(calls)
 
-    assert results == {"fedora": UpdateStatus.UPDATES_OK, "sd-app": UpdateStatus.UPDATES_REQUIRED}
+@pytest.mark.parametrize(
+    "templates,stderr,expected",
+    [
+        (
+            ["template"],
+            "template updating 0\ntemplate done success",
+            UpdateStatus.UPDATES_OK,
+        ),
+        (
+            ["template"],
+            "template updating 0\nunknown_keyword",
+            UpdateStatus.UPDATES_FAILED,
+        ),
+        (
+            ["tpl1", "tpl2"],
+            "tpl1 updating 0\ntpl2 updating 0\tpl1 done success\ntpl2 done error",
+            UpdateStatus.UPDATES_FAILED,
+        ),
+    ],
+)
+def test_apply_templates(templates, stderr, expected):
 
-    assert Updater.overall_update_status(results) == UpdateStatus.UPDATES_REQUIRED
-    assert not mocked_error.called
-    assert not apply_dom0.called
+    with mock.patch(
+        "sdw_updater.Updater._start_qubes_updater_proc",
+        return_value=subprocess.Popen(
+            "echo '{}' >> /dev/stderr".format(stderr),
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        ),
```

Comment:
I would consider mocking out subprocess here in Updater.py, instead of the helper method- a mock returning a running subprocess feels a bit unexpected (Even though it's just a test method, I'm also surprised that no linter caught the `shell=True` here - I guess we don't have bandit or similar enabled on this repo).  See also my comment on whether you may want this to be just a part of the main `apply_updates_templates` method rather than a helper.